### PR TITLE
paginator-swagger

### PR DESCRIPTION
**Pull request by OlehReex**

### CODE REVIEW

**Review by Alexsandr-Yablunovskyi — APPROVED** [review]
Nice work)

File: pom.xml [hunk 1]
```diff
@@ -44,6 +44,16 @@
             <artifactId>spring-boot-starter-test</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-validation</artifactId>
+            <version>3.1.5</version>
```
Comment (Elena-Bruyako) [inline pom.xml]: spring-boot-starter-parent already provide this version

File: src/main/java/onlinebookstore/exception/CustomGlobalExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+@ControllerAdvice
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(
+            MethodArgumentNotValidException ex,
+            HttpHeaders headers,
+            HttpStatusCode status,
+            WebRequest request) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("status", HttpStatus.BAD_REQUEST);
```
Comment (Elena-Bruyako) [inline src/main/java/onlinebookstore/exception/CustomGlobalExceptionHandler.java]: It's redundant to add status here, as you will have it in response

File: src/main/java/onlinebookstore/validation/IsbnValidator.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package onlinebookstore.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.regex.Pattern;
+
+public class IsbnValidator implements ConstraintValidator<Isbn, String> {
+    private static final String ISBN_PATTERN = "^\\d{2}-\\d{3}-\\d{3}-\\d{2}$";
```
Comment (Elena-Bruyako) [inline src/main/java/onlinebookstore/validation/IsbnValidator.java]:
```suggestion
    private static final Pattern ISBN_PATTERN = Pattern.compile("^\\d{2}-\\d{3}-\\d{3}-\\d{2}$");
```

File: src/main/java/onlinebookstore/controller/BookController.java [hunk 4]
```diff
@@ -1,48 +1,62 @@
...
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    @Operation(summary = "Get all books", description = "Get list of all books")
+    public List<BookDto> getAll(Pageable pageable) {
+        return bookService.findAll(pageable);
     }
 
     @GetMapping("/{id}")
+    @Operation(summary = "Get book", description = "Get book by id")
     public BookDto getBookById(@PathVariable Long id) {
```
Comment (Elena-Bruyako) [inline src/main/java/onlinebookstore/controller/BookController.java]: You can also validate @PathVariable by using @Positive and add @Validated annotation on your controller class (optional comment)

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's make your code better!

**Review by Andriy-dz — APPROVED** [review]

**Pull request merged**